build.gradle.kts

```diff
@@ -223,10 +227,29 @@ tasks {
 		options.compilerArgs.add("-Xlint:all,-exports")
 	}
 
+	// Prepares test-related JVM args
+	val moduleName = "org.junitpioneer"
+	val targetModule = if (modularBuild.toBoolean()) moduleName else "ALL-UNNAMED"
+	val patchModuleArg = "--patch-module=$moduleName=${compileJava.get().destinationDirectory.asFile.get().path}"
+	val testJvmArgs = listOf(
+			"-XX:+IgnoreUnrecognizedVMOptions",
+			"--add-opens=java.base/java.util=$targetModule",
+			"--add-opens=java.base/java.lang=$targetModule",
+			patchModuleArg
+	)
+
 	compileTestJava {
 		options.encoding = "UTF-8"
 		options.compilerArgs.add("-Werror")
-		options.compilerArgs.add("-Xlint:all")
+		var xlintArg = "-Xlint:all"
+		if (modularBuild.toBoolean()) {
+			options.compilerArgs.add(patchModuleArg)
+			xlintArg += ",-exports,-requires-automatic"
+			if (JavaVersion.current() >= JavaVersion.VERSION_16) {
+				xlintArg += ",-missing-explicit-ctor"
+			}
```

Comment:
Comments would be nice here, too. 😃

Reply:
Will do.

Fun fact: the key is missing in the official docs.

https://docs.oracle.com/en/java/javase/16/docs/specs/man/javac.html

But it was added in Java 16:

https://bugs.openjdk.org/browse/JDK-8071961